### PR TITLE
Fix enums bindings

### PR DESCRIPTION
**Pull request by maxim-sheronov**
Add missed bindings for enums
Move some enums to class to have correct output of api.json

### CODE REVIEW

**Comment by maxim-sheronov** [comment]
There is PR for #11183

**Comment by akien-mga** [comment]
CC @neikeq

**Comment by neikeq** [comment]
If @reduz agrees all enums should go into their actual class then this is good to merge (or maybe moving `PowerState` to `OS`, IDK). Otherwise, the solution I described [here](https://github.com/godotengine/godot/issues/11183#issuecomment-329122838) would be the way to go.

**Comment by maxim-sheronov** [comment]
I have moved PowerState enum to OS class. And fixed some static check errors.

**Review by karroffel — APPROVED** [review]

**Comment by karroffel** [comment]
You have some style issues, from what I see it looks good to go otherwise! 👌

**Comment by HummusSamurai** [comment]
Does this fix https://github.com/godotengine/godot/issues/11183?

File: core/bind/core_bind.cpp [hunk 1]
```diff
@@ -1142,11 +1142,11 @@ void _OS::_bind_methods() {
...
-	BIND_ENUM_CONSTANT(POWERSTATE_UNKNOWN);
-	BIND_ENUM_CONSTANT(POWERSTATE_ON_BATTERY);
-	BIND_ENUM_CONSTANT(POWERSTATE_NO_BATTERY);
-	BIND_ENUM_CONSTANT(POWERSTATE_CHARGING);
-	BIND_ENUM_CONSTANT(POWERSTATE_CHARGED);
+	BIND_ENUM_CONSTANT(OS::POWERSTATE_UNKNOWN);
+	BIND_ENUM_CONSTANT(OS::POWERSTATE_ON_BATTERY);
+	BIND_ENUM_CONSTANT(OS::POWERSTATE_NO_BATTERY);
+	BIND_ENUM_CONSTANT(OS::POWERSTATE_CHARGING);
+	BIND_ENUM_CONSTANT(OS::POWERSTATE_CHARGED);
```
Comment (neikeq) [inline core/bind/core_bind.cpp]: Sorry, didn't notice until now. This is wrong, the argument you pass to `BIND_CONSTANT` and `BIND_ENUM_CONSTANT` is stringified, therefore you cannot include `OS::` in it.

Comment (karroffel) [reply core/bind/core_bind.cpp]: Oh, should I revert or can you properly fix it?

Comment (neikeq) [reply core/bind/core_bind.cpp]: I am having a look at it right now.

Comment (neikeq) [reply core/bind/core_bind.cpp]: Looks like the way other classes do this is simply by declaring the same enum in `_OS` too. You can have a look at `SaverFlags` in `ResourceSaver` and `_ResourceSaver` as an example (BTW, the latter is missing the `FLAG_REPLACE_SUBRESOURCE_PATHS` constant).

cc @maxim-sheronov

Comment (neikeq) [reply core/bind/core_bind.cpp]: Here it is #11316

**Pull request merged**